invisible_cities/core/random_sampling.py

```diff
@@ -233,7 +233,7 @@ def dark_expectation(self, sample_width : int,
 
         pad_xbins, _ = pad_pdfs(self.xbins, self.probs)
 
-        if dark_model == DarkModel.threshold:
```

Comment:
What is the justification for switching == with "is"?  From what I see (e.g. https://realpython.com/courses/python-is-identity-vs-equality/), these are not identical comparisons.  In this case, the "threshold" attribute is probably just some number.  I don't know the process well enough to understand where it is assigned, but I can imagine it pointing to a different memory space than the dark_model variable.

Reply:
equality is an operation that takes two values and performs some logical operation on them, while identity compares memory positions. It is probably not critical in our case, but it is the right way to do it. Symbols are unique elements; thus, there is precisely one `DarkModel.threshold` (like there is only one `True` and one `False`), and you cannot make yourself another one.
I can think of two exapmles to showcase this:
- For `int`s, these two statements yield different results: `2**32 == 2**33/2`,  `2**32 is 2**33/2` (don't try it out with small integers, that's another story)
- In principle, I can write an object whose equality operator gives the wrong result. For instance
```
class A:
    def __eq__(self, other):
        return True
```
and then `A() == DarkModel.threshold` would return `True`. You cannot do the same with `is`.

regarding this:
>  the "threshold" attribute is probably just some number.

It's not. That's the associated value, which we mostly don't care about. That's the point of symbols, they are a completely separate entity. Effectively `DarkModel` is a new type and its attributes are the only possible values of this new type. Kind of like `bool` is a type and `True` and `False` are the only possible values of the `bool` type. Unlike the boolean type, `Enum`s have an associated value, but we (mostly) use them as symbols, where the associated value is meaningless and we only care about their uniqueness.

Reply:
Awesome, thanks.